app/logic/adminNewEvent.py

```diff
@@ -0,0 +1,63 @@
+from app.models.event import Event
+from datetime import *
+# from datetime import timedelta
+from app.models.facilitator import Facilitator
+# from dateutil import parser
+
+def calculateRecurringEventFrequency(recurringEventInfo):
+    """
+    """
+
+    eventName = recurringEventInfo['eventName']
+
+    endDate = datetime.strptime(recurringEventInfo['eventEndDate'], '%m/%d/%Y')
+    startDate = datetime.strptime(recurringEventInfo['eventStartDate'], '%m/%d/%Y')
+
+    recurringEvents = []
+
+    if endDate == startDate:
+        raise Exception("This event is not a recurring Event")
+
+    counter = 0
+    for i in range(0, ((endDate-startDate).days +1), 7):
+        counter += 1
+        recurringEvents.append({'eventName': f"{eventName} week {counter}",
+        'Date':startDate.strftime('%m/%d/%Y')})
+        startDate += timedelta(days=7)
+
+
+def setValueForUncheckedBox(eventData):
+
+    eventCheckBoxes = ['eventRequiredForProgram','eventRSVP', 'eventServiceHours', 'eventIsTraining', 'eventIsRecurring']
+
+    for checkBox in eventCheckBoxes:
+        if checkBox not in eventData:
+            eventData[checkBox] = False
+
+    return eventData
+
+def createNewEvent(newEventData):

```

Comment:
A function like this should really return the Event it has created, rather than a message that you want to be passed to the user. Let the controller worry about what message to create on success.